### PR TITLE
Added OTEL MeterProvider in Telemetery Settings

### PR DESCRIPTION
**Pull request by shinigami-777**
## Which problem is this PR solving?
- #6127

## Description of the changes
- This PR introduces `LeveledMeterProvider` in the  Telemetery Settings struct . This has been propagated to other parts in the codebase.

## How was this change tested?
- I had successfully run `make fmt` and `make lint test` for testing.

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

**Comment by mahadzaryab1** [comment]
@shinigami-777 The purpose of this ticket isn't necessarily to change how the settings are being initialized but rather to pass down the meter provider from OTEL. You can break down this task as follows: 
- [ ] Add the `LeveledMeterProvider` to [telemetry.Setting](https://github.com/jaegertracing/jaeger/blob/main/pkg/telemetery/settings.go#L16)
- [ ] Pass in the `LeveledMeterProvider` wherever it is available. For example, in the jaeger-v2 query server, you can pass it in [here](https://github.com/jaegertracing/jaeger/blob/main/cmd/jaeger/internal/extension/jaegerquery/server.go#L96) by doing `s.telset.LeveledMeterProvider`

**Comment by shinigami-777** [comment]
@mahadzaryab1 Thanks for the clarity. I have made the necessary changes.

File: pkg/telemetery/settings.go [hunk 1]
```diff
@@ -14,11 +16,12 @@ import (
...
-	TracerProvider trace.TracerProvider
-	Metrics        metrics.Factory
-	ReportStatus   func(*componentstatus.Event)
-	Host           component.Host
+	Logger               *zap.Logger
+	TracerProvider       trace.TracerProvider
+	Metrics              metrics.Factory
+	ReportStatus         func(*componentstatus.Event)
+	Host                 component.Host
+	LeveledMeterProvider func(_ configtelemetry.Level) metric.MeterProvider
```
Comment (mahadzaryab1) [inline pkg/telemetery/settings.go:24]: Now that we've got this field available to us - we also want to be able to use it. For example, in the [query server ](https://github.com/jaegertracing/jaeger/blob/main/cmd/query/app/server.go#L153)we currently hardcode the meter provider but now we can pass it from `telset`.

**Comment by shinigami-777** [comment]
@mahadzaryab1 I have propagated the setting everywhere in the codebase.

**Comment by yurishkuro** [comment]
solved by #6173

**Pull request closed without merging**